### DISCUSSION TITLE
Addition of Dialog Feature

### DISCUSSION THREAD

**Discussion by umarraza** [inertiajs/inertia#1839] [Ideas]
**Description**
Currently, when handling forms within our platform, we often find it convenient to open them in modal or dialog boxes. However, this approach presents certain limitations. For instance, consider a scenario where we need to submit student data through a dialog, which requires loading metadata into dropdowns. This necessitates loading the metadata via the index page and passing it to the modal component through props. Furthermore, the absence of browser history in this context poses additional challenges.

**Proposal**
I propose the addition of a dialog feature to Inertia. This feature is widely utilized in development and would significantly enhance the flexibility and efficiency of our workflows. By incorporating dialog functionality directly into Inertia, we can streamline form handling and improve the user experience.

**Comment by otherperspectives** [comment]
#642

**Comment by anchan42** [comment]
👍👍👍
I read some concerns over this feature but I think it will be useful to have.
People can choose not to use it if they don't like it, but at least they have the option.
